### PR TITLE
7 post api articles article id comments

### PR DESCRIPTION
**Pull request by layla93k**

### CODE REVIEW

**Review by robert112233 — CHANGES_REQUESTED** [review]
Cool looks good! 

A few missing tests: 

Status 404 username is not found in the database (psql will throw another error for you to catch) 

400 posting to an invalid id eg. notanumber

404 posting to a non-existent id eg. 9999999

Add these and submit another PR for review! Nice work ⭐️

File: Models/post-and-patch-models.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const { copy } = require('../app.js')
+const db = require('../db/connection.js')
+
+exports.insertNewComment = (newComment, articleId, author) => {
+    
+      if(isNaN(articleId)){
+        return Promise.reject({ status: 400, msg: 'bad request' })
+      }
```
Comment (itsaliabdallah) [inline Models/post-and-patch-models.js:8]: PSQL should throw an error if the articleID is not a valid number.

File: Models/post-and-patch-models.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    const query = `INSERT INTO comments(body, article_id, author)
+    VALUES ($1, $2, $3)
+    RETURNING*`
+    const maxArticleQuery = `SELECT MAX(article_id) FROM articles;`
```
Comment (itsaliabdallah) [inline Models/post-and-patch-models.js:12]: I'm not sure this part of your SQL query is needed or really doing anything right now?
If you make a request with an articleID which is a number and doesn't exist you should recieve an empty array.

File: Models/post-and-patch-models.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        if (articleId <= maxArticleId) {
+            return db.query(query, [newComment, articleId, author])
+                .then(({ rows }) => {
+                    return rows
+                })
+             } 
+            
+        return Promise.reject({ status: 404, msg: 'article_id does not exist' })
+                     
+    })
```
Comment (itsaliabdallah) [inline Models/post-and-patch-models.js:26]: As my previous comment about using MAX.
We shouldn't need to make two database queries.

**Review by itsaliabdallah — APPROVED** [review]
I like that you've thought about seperating your controllers out, maybe thing about seperating them out by article/comments etc instead of the method of the request.

Everything is looking nice, but please consider a refactor as we shouldn't need to have two database queries within a model.

**Pull request merged**